packages/fxa-settings/src/components/DropDownAvatarMenu/index.test.tsx

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { MockedCache } from '../../models/_mocks';
+import DropDownAvatarMenu from '.';
+
+describe('DropDownAvatarMenu', () => {
+  it('renders and toggles as expected with default values', () => {
+    render(
+      <MockedCache account={{ avatarUrl: null, displayName: null }}>
+        <DropDownAvatarMenu />
+      </MockedCache>
+    );
+
+    const toggleButton = screen.getByTestId('drop-down-avatar-menu-toggle');
+    const dropDownId = 'drop-down-avatar-menu';
+    const dropDown = screen.queryByTestId(dropDownId);
+
+    expect(toggleButton).toHaveAttribute('title', 'Firefox Account Menu');
+    expect(toggleButton).toHaveAttribute('aria-controls', dropDownId);
+    expect(toggleButton).toHaveAttribute('aria-expanded', 'false');
+    expect(dropDown).not.toBeInTheDocument;
+
+    fireEvent.click(toggleButton);
```

Comment:
huh, so we don't need to wait a turn for the event to propagate and the DOM to update? it looks like the [testing-library/react `fireEvent`](https://github.com/testing-library/react-testing-library/blob/master/src/fire-event.js) doesn't do anything special (setTimeout / rAF / wait to resolve a promise)

Reply:
It's magic!

Yeah, I'm not sure why we don't have to wait for a rerender in the test (it's nice we don't need to, guess it's automatic somehow), but we do have to wrap `fireEvent` in an `await act(async () => {` when we need to wait for a GQL mutation to run.

Reply:
huh, cool. maybe the test stays in virtual DOM-land 🤷